File: groundfire-0.25/src/shopmenu.cc
////////////////////////////////////////////////////////////////////////////////
//
//               Groundfire
//
////////////////////////////////////////////////////////////////////////////////
//
// Copyright (c) 2004, Tom Russell (tom@groundfire.net)
//
// This file is part of the Groundfire project, distributed under the MIT 
// license. See the file 'COPYING', included with this distribution, for a copy
// of the full MIT licence.
//
////////////////////////////////////////////////////////////////////////////////
//
//   File name : shopmenu.cc
//
//          By : Tom Russell
//
//        Date : 01-Jul-03
//
// Description : Handles the shopping menu which appears between rounds
//
//
//
////////////////////////////////////////////////////////////////////////////////

////////////////////////////////////////////////////////////////////////////////
// Includes
////////////////////////////////////////////////////////////////////////////////
#include "shopmenu.hh"

#include "tank.hh"
#include "weapon.hh"
#include "controls.hh"
#include "font.hh"

////////////////////////////////////////////////////////////////////////////////
// Public Member Functions
////////////////////////////////////////////////////////////////////////////////

////////////////////////////////////////////////////////////////////////////////
//
// Function    : cShopMenu
//
// Description : Constructor
//
////////////////////////////////////////////////////////////////////////////////
cShopMenu::cShopMenu
(
    cGame * game
)
: cMenu (game)
{
    for (int i = 0; i < 8; i++) 
    {
        // Start everyone with a small delay incase someone was pressing fire 
        // upon entry to the shopping screen.
        _playerSelectPos[i]   = 0;
        _playerSelectDelay[i] = 0.4f;
        _playerDone[i]        = false;
    }

    // 'linelit' is to prevent us drawing too many transparent bars across one 
    // option.
    for (int i = 0; i < 10; i++) 
    {
        _lineLit[i] = false;
    }

    _jumpjetsCost = game->getSettings ()->getInt ("Price", "Jumpjets", 50);

}

////////////////////////////////////////////////////////////////////////////////
//
// Function    : ~cShopMenu
//
// Description : Destructor
//
////////////////////////////////////////////////////////////////////////////////
cShopMenu::~cShopMenu
(
)
{
}

////////////////////////////////////////////////////////////////////////////////
//
// Function    : update
//
// Description : Update the shopping menu
//
////////////////////////////////////////////////////////////////////////////////
enumGameState
cShopMenu::update
(
    double time
)
{
    updateBackground (time);

    cPlayer ** players = _game->getPlayers ();

    bool stillPlayersInShop = false;

    for (int i = 0; i < 8; i++) 
    {
        if (players[i] != NULL && _playerDone[i] == false)
        {            
            stillPlayersInShop = true;

            // Slow down the rate at which the user can scroll down the list by
            // pausing for a short delay after every move
            if (_playerSelectDelay[i] < 0.0f) 
            {
                players[i]->update ();

                double when; // Not used here

                if (players[i]->getCommand (CMD_GUNUP, when))
                {
                    _playerSelectPos[i]--;

                    if (_playerSelectPos[i] == -1) 
                    {
                        _playerSelectPos[i] = 10;
                    }

                    _playerSelectDelay[i] = 0.2f;
                }
                else if (players[i]->getCommand (CMD_GUNDOWN, when))
                {
                    _playerSelectPos[i]++;
                    
                    if (_playerSelectPos[i] == 11) 
                    {
                        _playerSelectPos[i] = 0;
                    }

                    _playerSelectDelay[i] = 0.2f;
                }
                else if (players[i]->getCommand (CMD_FIRE, when))
                {
                    // This tank is trying to purchase a weapon!
                    cTank * tank = players[i]->getTank ();

                    switch (_playerSelectPos[i])
                    {
                    case 0: // Machine Gun
                        if (   players[i]->_money 
                            >= tank->_weapon[MACHINEGUN]->getCost ())
                        {
                            players[i]->_money -= 
                                tank->_weapon[MACHINEGUN]->getCost ();

                            tank->_weapon[MACHINEGUN]->addAmount (50);
                        }
                        break;

                    case 1: // Jump Jets 
                        // Check funds
                        if (players[i]->_money >= _jumpjetsCost)
                        {
                            players[i]->_money -= _jumpjetsCost;
                            tank->_totalFuel += 1.0f;
                        }
                        break;

                    case 2: // Mirvs
                        //check funds
                        if (   players[i]->_money 
                            >= tank->_weapon[MIRVS]->getCost ())
                        {
                            players[i]->_money -= 
                                tank->_weapon[MIRVS]->getCost ();

                            tank->_weapon[MIRVS]->addAmount (1);
                        }
                        break;
                        
                    case 3: // Missiles 
                        // check funds
                        if (   players[i]->_money
                            >= tank->_weapon[MISSILES]->getCost ())
                        {
                            // missiles come in packs of 5
                            players[i]->_money -= 
                                tank->_weapon[MISSILES]->getCost ();

                            tank->_weapon[MISSILES]->addAmount (5); 
                        }
                        break;
                        
                    case 4: // Nukes
                        //check funds
                        if (   players[i]->_money 
                            >= tank->_weapon[NUKES]->getCost ())
                        {
                            players[i]->_money -= 
                                tank->_weapon[NUKES]->getCost ();

                            tank->_weapon[NUKES]->addAmount (1);
                        }
                        break;
                        
                        // These aren't implemented yet!
                    case 5: // Rolling Mines
                    case 6: // Airstrike
                    case 7: // Death's Head
                    case 8: // Hover Coil
                    case 9: // Corbomite
                        break;

                        // We've finished 
                    case 10:
                        _playerDone[i] = true;
                        break;
                        
                    default:
                        break;
                    }

                    // The delay here is quite important. We don't want anyone 
                    // to accidentally buying multiple times
                    _playerSelectDelay[i] = 0.2f;    
                }
            }
            else
            {
                _playerSelectDelay[i] -= time;
            }

            _lineLit[_playerSelectPos[i]] = true;
        }
    }

    // Has everyone finished shopping?
    if (stillPlayersInShop)
    {
        return (CURRENT_STATE);
    }
    else
    {
        return (ROUND_STARTING);
    }
}

////////////////////////////////////////////////////////////////////////////////
//
// Function    : draw
//
// Description : Draw the menu
//
////////////////////////////////////////////////////////////////////////////////
void
cShopMenu::draw
(
)
{
    drawBackground ();

    glEnable (GL_BLEND);
  
    glColor4f (0.6f, 0.3f, 0.0f, 0.5f);

    cPlayer ** players = _game->getPlayers ();

    for (int i = 0; i < 8; i++)
    {
        if (players[i] != NULL && _playerDone[i] == false)
        {
            glBegin (GL_QUADS);
            
            glColor4f (0.3f, 0.1f, 0.1f, 0.5f);

            // Draw the vertical bars for each tank. The tabs showing the 
            // current funds appear at either the top or bottom on alternate 
            // tanks.

            if (i % 2 == 0)
            {
                glVertex3f   (-9.4f + (i * 1.5f),  5.6f, 0.0f);
                glVertex3f   (-9.4f + (i * 1.5f),  4.9f, 0.0f);
                glVertex3f   (-6.6f + (i * 1.5f),  4.9f, 0.0f);
                glVertex3f   (-6.6f + (i * 1.5f),  5.6f, 0.0f);

                glVertex3f   (-9.4f + (i * 1.5f), -5.3f, 0.0f);
                glVertex3f   (-9.4f + (i * 1.5f),  4.9f, 0.0f);
                glVertex3f   (-8.1f + (i * 1.5f),  4.9f, 0.0f);
                glVertex3f   (-8.1f + (i * 1.5f), -5.3f, 0.0f);

            }
            else
            {
                glVertex3f   (-10.9f + (i * 1.5f), -5.5f, 0.0f);
                glVertex3f   (-10.9f + (i * 1.5f), -6.2f, 0.0f);
                glVertex3f   (-8.1f  + (i * 1.5f), -6.2f, 0.0f);
                glVertex3f   (-8.1f  + (i * 1.5f), -5.5f, 0.0f);

                glVertex3f   (-9.4f + (i * 1.5f), -5.5f, 0.0f);
                glVertex3f   (-9.4f + (i * 1.5f),  4.7f, 0.0f);
                glVertex3f   (-8.1f + (i * 1.5f),  4.7f, 0.0f);
                glVertex3f   (-8.1f + (i * 1.5f), -5.5f, 0.0f);
            }
            
            // if the weapons line the tank is on is already lit, don't light 
            // it again. This prevents it becoming saturated.
            if (_lineLit[_playerSelectPos[i]] == true)
            {
                float v = _playerSelectPos[i] * 0.8f;

                glColor4f (1.0f, 1.0f, 1.0f, 0.1f);
                glVertex3f   (-9.4f, 3.8f - v, 0.0f);
                glVertex3f   (-9.4f, 4.6f - v, 0.0f);
                glVertex3f   ( 9.4f, 4.6f - v, 0.0f);
                glVertex3f   ( 9.4f, 3.8f - v, 0.0f);

                _lineLit[_playerSelectPos[i]] = false;
            }
            
            glEnd ();
        }
    }

    glDisable (GL_BLEND);

    _font->setSize (0.3f, 0.3f, 0.25f);
    _font->setColour (1.0f, 1.0f, 1.0f);

    for (int i = 0; i < 8; i++) 
    {
        if (players[i] != NULL && _playerDone[i] == false)
        {
            float r, g, b;

            cTank * tank = players[i]->getTank ();

            tank->getColour (r, g, b);

            // Draw a graphic of each tank
            glColor3f (r, g, b);

            float v = _playerSelectPos[i] * 0.8f;

            glLoadIdentity ();

            glBegin (GL_QUADS);

            glVertex3f   (-9.35f + (i * 1.5f), 3.9f - v, 0.0f);
            glVertex3f   (-9.05f + (i * 1.5f), 4.5f - v, 0.0f);
            glVertex3f   (-8.45f + (i * 1.5f), 4.5f - v, 0.0f);
            glVertex3f   (-8.15f + (i * 1.5f), 3.9f - v, 0.0f);

            glEnd ();    

            // Below the tank graphic, draw an indicator of the current amount 
            // of the weapon we are currently lined up with. This is either a 
            // bar (for jumpjets) or a number representing the current 
            // inventory (e.g. nukes)

            switch (_playerSelectPos[i])
            {
            case 0: // Machine Gun
                drawBars (-9.4f + (i * 1.5f), 3.7f - v, 
                          tank->_weapon[MACHINEGUN]->getAmmo () / 50.0f);
                break;

            case 1: // Jump Jets
                drawBars (-9.4f + (i * 1.5f), 3.7f - v, tank->_totalFuel);
                break;

            case 2: // Mirvs
                _font->printCentredAt (-8.75 + (i * 1.5f), 3.5f - v, "x%d", 
                                       tank->_weapon[MIRVS]->getAmmo ());
                break;

            case 3: // Missiles
                _font->printCentredAt (-8.75 + (i * 1.5f), 3.5f - v, "x%d", 
                                       tank->_weapon[MISSILES]->getAmmo ());
                break;

            case 4: // Nukes
                _font->printCentredAt (-8.75 + (i * 1.5f), 3.5f - v, "x%d", 
                                       tank->_weapon[NUKES]->getAmmo ());
                break;

            case 5: // Rolling Mines
            case 6: // Airstrike
            case 7: // Death's Head
            case 8: // Hover Coil
            case 9: // Corbomite
                break;

            default:
                break;
            }
        }
    }

    _font->setShadow (true);
    // Write the number of the round we're about to start.
    _font->setSize (0.6f, 0.6f, 0.5f);
    _font->setColour (1.0f, 1.0f, 1.0f);

    _font->printCentredAt (0.0f, 6.5f, "Round %d of %d", 
                           _game->getCurrentRound () + 1, 
                           _game->getNumOfRounds ());

    _font->setSize (0.4f, 0.4f, 0.3f);
    _font->setColour (0.9f, 0.9f, 0.9f);
    
    int nukeCost       = players[0]->getTank ()->_weapon[NUKES]->getCost ();
    int missileCost    = players[0]->getTank ()->_weapon[MISSILES]->getCost ();
    int mirvCost       = players[0]->getTank ()->_weapon[MIRVS]->getCost ();
    int machineGunCost = players[0]->getTank ()->_weapon[MACHINEGUN]->getCost();

    // Write the weapon names for each row.

    _font->printCentredAt (4.0f,  5.0f, "Cost");
    _font->printCentredAt (7.0f,  5.0f, "Item");

    _font->printCentredAt (7.0f,  4.0f, "Machine Gun");
    _font->printCentredAt (4.0f,  4.0f, "$%d", machineGunCost);
    _font->printCentredAt (7.0f,  3.2f, "Jump Jet");
    _font->printCentredAt (4.0f,  3.2f, "$%d", _jumpjetsCost);
    _font->printCentredAt (7.0f,  2.4f, "Mirvs");
    _font->printCentredAt (4.0f,  2.4f, "$%d", mirvCost);
    _font->printCentredAt (7.0f,  1.6f, "Missiles");
    _font->printCentredAt (4.0f,  1.6f, "$%d", missileCost);
    _font->printCentredAt (7.0f,  0.8f, "Nukes");
    _font->printCentredAt (4.0f,  0.8f, "$%d", nukeCost);
    _font->setColour (0.3f, 0.3f, 0.3f);
    _font->printCentredAt (7.0f,  0.0f, "Rolling Mines");
    _font->printCentredAt (4.0f,  0.0f, "$50");
    _font->printCentredAt (7.0f, -0.8f, "Airstrike");
    _font->printCentredAt (4.0f, -0.8f, "$100");
    _font->printCentredAt (7.0f, -1.6f, "Death's Head");
    _font->printCentredAt (4.0f, -1.6f, "$200");
    _font->printCentredAt (7.0f, -2.4f, "Hover Coil");
    _font->printCentredAt (4.0f, -2.4f, "$150");
    _font->printCentredAt (7.0f, -3.2f, "Corbomite");
    _font->printCentredAt (4.0f, -3.2f, "$20");

    _font->setColour (0.9f, 0.9f, 0.9f);
    _font->printCentredAt (7.0f, -4.0f, "Done!");

    _font->setSize (0.35f, 0.35f, 0.275f);

    // Write the amount of money everybody has
    for (int i = 0; i < 8; i++)
    {
        if (players[i] != NULL && _playerDone[i] == false)
        {
            if (i % 2 == 0) 
            {
                _font->printCentredAt (-8.0f + (i * 1.5f), 5.1f, 
                               "$%d", players[i]->_money);
            }
            else
            {
                _font->printCentredAt (-9.5f + (i * 1.5f), -6.0f, 
                               "$%d", players[i]->_money);    
            }
        }
    }
    _font->setShadow (false);
}

////////////////////////////////////////////////////////////////////////////////
//
// Function    : drawBars
//
// Description : Draw the bars representing inventory (of things like jumpjet 
//               fuel)
//
////////////////////////////////////////////////////////////////////////////////
void
cShopMenu::drawBars
(
    float x,
    float y,
    float numberOfBars
)
{
    glColor3f (1.0f, 1.0f, 1.0f);

    for (int i = 0; numberOfBars > 0.0f; i++)
    {
        float lengthOfBar;

        if (numberOfBars > 1.0f)
        {
            lengthOfBar = 1.3f;
        }
        else
        {
            lengthOfBar = 1.3f * numberOfBars;
        }
        numberOfBars -= 1.0f;

        glBegin (GL_QUADS);

        glVertex3f   (x,               y - (0.2f * i) + 0.15f, 0.0f);
        glVertex3f   (x,               y - (0.2f * i),         0.0f);
        glVertex3f   (x + lengthOfBar, y - (0.2f * i),         0.0f);
        glVertex3f   (x + lengthOfBar, y - (0.2f * i) + 0.15f, 0.0f);

        glEnd ();  
    } 
}

File: versao-python/src/shopmenu.py
from typing import TYPE_CHECKING
from .menu import Menu
from .player import Player
from .tank import Tank
from .common import GameState
import pygame

if TYPE_CHECKING:
    from .game import Game

class ShopMenu(Menu):
    def __init__(self, game: 'Game'):
        super().__init__(game)
        
        self._player_select_pos = [0] * 8
        self._player_select_delay = [0.4] * 8
        self._player_done = [False] * 8
        self._line_lit = [False] * 11
        
        settings = game.get_settings()
        self._jumpjets_cost = settings.get_int("Price", "Jumpjets", 50)
        
    def update(self, time: float) -> int:
        self.update_background(time)
        
        players = self._game.get_players()
        still_players_in_shop = False
        
        for i in range(8):
            if players[i] is not None and not self._player_done[i]:
                still_players_in_shop = True
                
                if self._player_select_delay[i] < 0.0:
                    players[i].update()
                    
                    if players[i].get_command(Player.CMD_GUNUP):
                        self._player_select_pos[i] -= 1
                        if self._player_select_pos[i] == -1: self._player_select_pos[i] = 10
                        self._player_select_delay[i] = 0.2
                        
                    elif players[i].get_command(Player.CMD_GUNDOWN):
                        self._player_select_pos[i] += 1
                        if self._player_select_pos[i] == 11: self._player_select_pos[i] = 0
                        self._player_select_delay[i] = 0.2
                        
                    elif players[i].get_command(Player.CMD_FIRE):
                         tank = players[i].get_tank()
                         pos = self._player_select_pos[i]
                         money = players[i].get_money()
                         
                         if pos == 0:
                             cost = tank.get_weapon(Tank.MACHINEGUN).get_cost()
                             if money >= cost:
                                 players[i].set_money(money - cost)
                                 tank.get_weapon(Tank.MACHINEGUN).add_amount(50)
                         elif pos == 1:
                             if money >= self._jumpjets_cost:
                                 players[i].set_money(money - self._jumpjets_cost)
                                 tank.set_total_fuel(tank.get_total_fuel() + 1.0)
                         elif pos == 2:
                             cost = tank.get_weapon(Tank.MIRVS).get_cost()
                             if money >= cost:
                                 players[i].set_money(money - cost)
                                 tank.get_weapon(Tank.MIRVS).add_amount(1)
                         elif pos == 3:
                             cost = tank.get_weapon(Tank.MISSILES).get_cost()
                             if money >= cost:
                                 players[i].set_money(money - cost)
                                 tank.get_weapon(Tank.MISSILES).add_amount(5)
                         elif pos == 4:
                             cost = tank.get_weapon(Tank.NUKES).get_cost()
                             if money >= cost:
                                 players[i].set_money(money - cost)
                                 tank.get_weapon(Tank.NUKES).add_amount(1)
                         elif pos == 10:
                             self._player_done[i] = True
                             
                         self._player_select_delay[i] = 0.2
                else:
                    self._player_select_delay[i] -= time
                    
                self._line_lit[self._player_select_pos[i]] = True
                
        if still_players_in_shop:
            return GameState.CURRENT_STATE
        else:
            return GameState.ROUND_STARTING

    def draw(self):
        self.draw_background()
        
        interface = self._game.get_interface()
        players = self._game.get_players()
        
        for i in range(8):
            if players[i] is not None and not self._player_done[i]:
                bar_col = (76, 25, 25, 128)
                polys = []
                if i % 2 == 0:
                     polys.append([(-9.4 + i*1.5, 5.6), (-9.4 + i*1.5, 4.9), (-6.6 + i*1.5, 4.9), (-6.6 + i*1.5, 5.6)])
                     polys.append([(-9.4 + i*1.5, -5.3), (-9.4 + i*1.5, 4.9), (-8.1 + i*1.5, 4.9), (-8.1 + i*1.5, -5.3)])
                else:
                    polys.append([(-10.9 + i*1.5, -5.5), (-10.9 + i*1.5, -6.2), (-8.1 + i*1.5, -6.2), (-8.1 + i*1.5, -5.5)])
                    polys.append([(-9.4 + i*1.5, -5.5), (-9.4 + i*1.5, 4.7), (-8.1 + i*1.5, 4.7), (-8.1 + i*1.5, -5.5)])
                    
                for pts in polys:
                    self.draw_game_polygon(pts, bar_col)
                    
                pos = self._player_select_pos[i]
                if self._line_lit[pos]:
                     v = pos * 0.8
                     high_col = (255, 255, 255, 25)
                     h_pts = [(-9.4, 3.8 - v), (-9.4, 4.6 - v), (9.4, 4.6 - v), (9.4, 3.8 - v)]
                     self.draw_game_polygon(h_pts, high_col)
                     self._line_lit[pos] = False

        for i in range(8):
            if players[i] is not None and not self._player_done[i]:
                 tank = players[i].get_tank()
                 c = tank.get_colour()
                 v = self._player_select_pos[i] * 0.8
                 t_pts = [(-9.35 + i*1.5, 3.9 - v), (-9.05 + i*1.5, 4.5 - v), (-8.45 + i*1.5, 4.5 - v), (-8.15 + i*1.5, 3.9 - v)]
                 self.draw_game_polygon(t_pts, c)
                 
                 pos = self._player_select_pos[i]
                 cx = -9.4 + i*1.5
                 cy = 3.7 - v

                 if pos == 0: self._draw_bars(cx, cy, tank.get_weapon(Tank.MACHINEGUN).get_ammo() / 50.0)
                 elif pos == 1: self._draw_bars(cx, cy, tank.get_total_fuel())
                 elif pos == 2: self._ui.draw_centered_text(cx + 0.65, cy - 0.2, f"x{tank.get_weapon(Tank.MIRVS).get_ammo()}", style=self._ui.style(0.3, (255, 255, 255), spacing=0.25))
                 elif pos == 3: self._ui.draw_centered_text(cx + 0.65, cy - 0.2, f"x{tank.get_weapon(Tank.MISSILES).get_ammo()}", style=self._ui.style(0.3, (255, 255, 255), spacing=0.25))
                 elif pos == 4: self._ui.draw_centered_text(cx + 0.65, cy - 0.2, f"x{tank.get_weapon(Tank.NUKES).get_ammo()}", style=self._ui.style(0.3, (255, 255, 255), spacing=0.25))

        self._ui.draw_centered_text(
            0.0,
            6.5,
            f"Round {self._game.get_current_round() + 1} of {self._game.get_num_of_rounds()}",
            style=self._ui.style(0.6, (255, 255, 255), shadow=True),
        )
        
        ref_tank = None
        for p in players:
             if p: 
                 ref_tank = p.get_tank()
                 break
        
        if ref_tank:
            nuke_cost = ref_tank.get_weapon(Tank.NUKES).get_cost()
            missile_cost = ref_tank.get_weapon(Tank.MISSILES).get_cost()
            mirv_cost = ref_tank.get_weapon(Tank.MIRVS).get_cost()
            mg_cost = ref_tank.get_weapon(Tank.MACHINEGUN).get_cost()
        else:
            nuke_cost = 0; missile_cost = 0; mirv_cost = 0; mg_cost = 0

        bright_style = self._ui.style(0.4, (230, 230, 230), spacing=0.3, shadow=True)
        self._ui.draw_centered_text(4.0, 5.0, "Cost", style=bright_style)
        self._ui.draw_centered_text(7.0, 5.0, "Item", style=bright_style)
        
        items = [("Machine Gun", mg_cost), ("Jump Jet", self._jumpjets_cost), ("Mirvs", mirv_cost), ("Missiles", missile_cost), ("Nukes", nuke_cost)]
        y = 4.0
        for name, cost in items:
            self._ui.draw_centered_text(7.0, y, name, style=bright_style)
            self._ui.draw_centered_text(4.0, y, f"${cost}", style=bright_style)
            y -= 0.8
            
        muted_style = self._ui.style(0.4, (76, 76, 76), spacing=0.3, shadow=True)
        gray_items = [("Rolling Mines", 50), ("Airstrike", 100), ("Death's Head", 200), ("Hover Coil", 150), ("Corbomite", 20)]
        for name, cost in gray_items:
             self._ui.draw_centered_text(7.0, y, name, style=muted_style)
             self._ui.draw_centered_text(4.0, y, f"${cost}", style=muted_style)
             y -= 0.8
             
        self._ui.draw_centered_text(7.0, -4.0, "Done!", style=bright_style)

        money_style = self._ui.style(0.35, (230, 230, 230), spacing=0.275, shadow=True)
        for i in range(8):
             if players[i] is not None and not self._player_done[i]:
                 if i % 2 == 0:
                     self._ui.draw_centered_text(-8.0 + i*1.5, 5.1, f"${players[i].get_money()}", style=money_style)
                 else:
                     self._ui.draw_centered_text(-9.5 + i*1.5, -6.0, f"${players[i].get_money()}", style=money_style)

    def _draw_bars(self, x: float, y: float, num_bars: float):
        interface = self._game.get_interface()
        col = (255, 255, 255)
        i = 0
        while num_bars > 0.0:
            length = 1.3 if num_bars > 1.0 else 1.3 * num_bars
            num_bars -= 1.0
            pts = [(x, y - 0.2*i + 0.15), (x, y - 0.2*i), (x + length, y - 0.2*i), (x + length, y - 0.2*i + 0.15)]
            self.draw_game_polygon(pts, col)
            i += 1
